### ISSUE TITLE
SpiGot/Craftbukkit API update for minecraft 1.13

### ISSUE THREAD

**Issue by Pingger** [RegalOwl/HyperConomy#143]
Hi,

the SpiGot API is being updated to the new Minecraft version, which invalidates large amounts of the code. (https://www.spigotmc.org/threads/bukkit-craftbukkit-spigot-bungeecord-1-13-pre7-development-builds.328883/) 
There is also quite an amount of new Items to support.

Are going to update HC to 1.13? If yes, any ETA?

~~I'd help, but I can't get eclipse to properly load the project, because of the dependencies (which can't be updated or fixed for some reason)~~ Got it working see PR

Corresponding PR: #149 

Greetings

**Comment by patrick-layden** [comment]
Hi, thanks for creating the ticket.  It's nice to see that my plugin is still being used.  Life has been busy recently, but I'll take a look and see if I can get an update out.  

The dependency issues are an easy fix; my Maven repo needs to be fixed. 

But if the update has broken all existing item definitions and some of the low-level code, the update might be pretty involved.  So no ETA, but I'll take a look and see what I can do.

**Comment by roracle** [comment]
Yeah even whoever updated the code about a year ago forgot to include a number of things like Nether Wart Blocks, Lava Blocks, and probably others. I've recently been trying to set up my economy with Hyperconomy but with some blocks not recognized there's almost no reason.

I love this plugin, there's no other like it. It keeps competition going good because it helps people price things in a competitive manner. (I let players set prices in their own shops an which are taxed, and the taxes go to the server held account).

I recently got md_5 to help me compile the updated code into a jar because the one available was still a version below. That's how I know that his item update didn't get all the items. 

Question though: you say not to use essentials' economy setting. So if I used Hyperconomy built in economy, can players still trade without having to rely on Hyperconomy shops etc?

**Comment by Pingger** [comment]
Ok, got it to compile on 1.12.2, and now I changed the dependency to a local 1.13 spigot jar and source. 
I'm now looking into the compile errors that arise from the new API and see if I can create a PR for that, when I'm done

Greetings


@roracle Since HyperConomy is Vault integrated, you should be able to use the economy of HC and any shop you like. On my server we used actually ChestShops, that connected to HC.

**Comment by roracle** [comment]
Which chest shop is that? So many out there and a few good ones with that name

**Comment by Pingger** [comment]
https://dev.bukkit.org/projects/chestshop?gameCategorySlug=bukkit-plugins&projectID=31263

**Comment by frost-byte** [comment]
@Pingger  Are you still actively working on this?

**Comment by Pingger** [comment]
@frost-byte I'm waiting for @RegalOwl to review my Pull Request, so that I know, that I can continue my work. There were some big changes in spigot from 1.12 to 1.13, that require a bigger rework at some Code Locations

see #149

**Comment by frost-byte** [comment]
Yeah, I totally understand that, I've been doing similar things with other plugins and the ones I'm working on.

**Comment by patrick-layden** [comment]
Sorry it's taking me so long to get to this.  I haven't been home with access to my computer much this summer and I'll be away for a few more weeks.  Feel free to distribute builds with your code to people that want it if it's working.  When I'm back I'll see if I can get this updated and merged.

**Comment by Pingger** [comment]
It is not fully working yet. Several Items are not in the Database, not all potions work, BUT it compiles and boots up.

**Comment by Pingger** [comment]
I just uploaded a bleeding edge development build: https://www.iskariot.info/jenkins/job/HyperConomy/job/HyperConomy%20by%20Pingger/
Not all potions are working!  
Enchantments are highly experimental!  
Playerheads/Skulls are uncertain!  
There are not few items that aren't working properly! (e.g. everything out of gold, because of a rename "gold_shovel" to "golden_shovel" for example)  
Database compatibility is NOT GUARANTEED!  

**MAKE BACKUPS! THIS NOT AN OFFICIAL RELEASE, But it boots on a clean server**

Issues can be reported in the following fork: https://github.com/Pingger/HyperConomy

**Comment by Pingger** [comment]
I just setup a build job for the HyperConomy plugin: https://www.iskariot.info/jenkins/job/HyperConomy/job/HyperConomy%20by%20Pingger/
